components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountDisableHandler.java

```diff
@@ -386,6 +408,48 @@ private void setUserClaim(String claimName, String claimValue, UserStoreManager
         }
     }
 
+    private void publishPreAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map) throws
+            AccountLockException {
+
+        AccountUtil.publishEvent(accountDisabledEventName, AccountUtil.cloneMap(map));
+    }
+
+    private void publishPostAccountDisabledEvent(String accountDisabledEventName, Map<String, Object> map, boolean
+            isDisblePropertySuccessfullyModified) throws AccountLockException {
+
+        Map<String, Object> eventProperties = AccountUtil.cloneMap(map);
+        eventProperties.put(IdentityEventConstants.EventProperty.UPDATED_DISABLED_STATUS,
```

Comment:
If `eventProperties` map in null, it will cause a NPE. Let's add a MapUtils.isNotEmpty to avoid this.

Reply:
It is already handled from cloneMap method.

Reply:
cloneMap method can return a null value right?